### PR TITLE
Exchange active_fe_index information for p::d::Triangulation.

### PR DESCRIPTION
**Pull request by bangerth**
This patch makes sure that we exchange the `active_fe_index` information on ghost cells for `hp::DoFHandler` objects on `p::d::Triangulation`. (#4712 does this for `p::shared::Triangulation`.) It uses the facilities of #4788 to do the ghost exchange.

I don't have a test that I can add to this PR because this functionality would only be exercised when really doing hp on `p::d::Triangulation`, which is not completed yet. But I have a few tests that I've been running on a branch that do that at least to the point where they fail (after the current function has run), so I think it must at least be mostly correct. There will of course be many tests that will exercise this code path -- everything that uses hp with p::d. For the moment, I just want to get this intermediate piece of functionality in so I can move to whatever fails next.

In reference to #3511.

### CODE REVIEW

File: source/hp/dof_handler.cc [hunk 1]
```diff
@@ -912,7 +913,26 @@ namespace internal
           else if (const parallel::distributed::Triangulation<dim, spacedim> *tr =
                      dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim>*> (&dof_handler.get_triangulation()))
             {
-              Assert (false, ExcNotImplemented());
+              // for completely distributed meshes, use the function that is able to move
+              // data from locally owned cells on one processor to the corresponding
+              // ghost cells on others. to this end, we need to have functions that can pack
```
Comment (davydden) [inline source/hp/dof_handler.cc]: To this end...

File: source/hp/dof_handler.cc [hunk 2]
```diff
@@ -912,7 +913,26 @@ namespace internal
...
+              // up and unpack again the data we want to transport -- namely, the
```
Comment (davydden) [inline source/hp/dof_handler.cc]: "pack and unpack", maybe?

**Comment by bangerth** [comment]
I made the requested comment changes.

**Comment by tjhei** [comment]
Alright, I am okay with making an exception and accepting it without a test. ;-)

**Comment by bangerth** [comment]
Fixes #4638.

**Pull request merged**